assembl/static2/js/app/pages/brightMirrorFiction.jsx

```diff
@@ -6,91 +6,207 @@ import { I18n } from 'react-redux-i18n';
+  fictionId: string
+};
+
+type BrightMirrorFictionReduxProps = {
+  /** Fiction locale fetched from mapStateToProps */
+  contentLocale: string,
+  /** Fiction locale mapping fetched from mapStateToProps */
+  contentLocaleMapping: any
+};
+
+export type BrightMirrorFictionData = {
+  /** Fiction object formatted through GraphQL  */
+  fiction: BrightMirrorFictionFragment,
+  /** GraphQL flag that checks the query/mutation state */
+  loading: boolean,
+  /** GraphQL error object used to handle fetching errors */
+  error: any
 };
 
-type State = {
+export type IdeaWithCommentsData = {
+  /**  Idea object formatted through GraphQL */
+  idea: IdeaWithCommentsQuery,
+  /** GraphQL flag that checks the query/mutation state */
+  loading: boolean,
+  /** GraphQL error object used to handle fetching errors */
+  error: any,
+  /** GraphQL refetch function in order to refresh data once a comment is submitted */
+  refetch: Function
+};
+
+type BrightMirrorFictionGraphQLProps = {
+  /** Fiction data information fetched from GraphQL */
+  brightMirrorFictionData: BrightMirrorFictionData,
+  /** Create comment mutation from GraphQL */
+  createComment: Function,
+  /** Fiction data information fetched from GraphQL */
+  ideaWithCommentsData: IdeaWithCommentsData
+};
+
+// Type use for creating a Bright Mirror comment with CreateCommentMutation
+type CreateCommentInputs = {
+  /** Comment body content */
+  body: string,
+  /** Comment content locale */
+  contentLocale: string,
+  /** Comment idea identifier */
+  ideaId: string,
+  /** Comment parent identifier */
+  parentId: string
+};
+
+type LocalBrightMirrorFictionProps = BrightMirrorFictionProps & BrightMirrorFictionReduxProps & BrightMirrorFictionGraphQLProps;
+
+type BrightMirrorFictionState = {
   /** Fiction title */
   title: string,
   /** Fiction content */
   content: string,
+  /** GraphQL loading flag */
+  loading: boolean,
   /** Fiction publication state */
   publicationState: string
 };
 
-const EMPTY_STRING = '';
-
-export class BrightMirrorFiction extends Component<Props, State> {
-  constructor(props: Props) {
+export class BrightMirrorFiction extends Component<LocalBrightMirrorFictionProps, BrightMirrorFictionState> {
+  // Lifecycle functions
+  constructor(props: LocalBrightMirrorFictionProps) {
     super(props);
     this.state = {
-      title: props.data.fiction.subject || EMPTY_STRING,
-      content: props.data.fiction.body || EMPTY_STRING,
-      publicationState: props.data.fiction.publicationState || PublicationStates.PUBLISHED
+      title: EMPTY_STRING,
+      content: EMPTY_STRING,
+      loading: props.brightMirrorFictionData.loading,
+      publicationState: PublicationStates.PUBLISHED
     };
   }
 
+  componentWillReceiveProps(nextProps: LocalBrightMirrorFictionProps) {
+    // Sync state
+    this.setState({
+      title: nextProps.brightMirrorFictionData.fiction.subject ? nextProps.brightMirrorFictionData.fiction.subject : EMPTY_STRING,
```

Comment:
Please destructure everything from nextProps.brightMirrorFictionData.fiction here for readability :)